tabpfn_client/config.py

```diff
@@ -30,6 +33,8 @@ def init(use_server=True):
         return
 
     if use_server:
+        ServiceClient.set_module_name(module_name)
```

Comment:
You are not expecting this to change during service run. It should be ONLY in the constructor of ServiceClient and then stored as a private member.

Reply:
ServiceClient is a singleton class, hence we don't have an constructor as per say.
I'm also wondering if there's a better design for classes like `ServiceClient`

Reply:
There should be. Singletons are to be handled with cautions, because as for global variables, they can be changed in multiple places and thus introduce side-effects. Also they are harder to test, because dependency injection does not work.

I'd need to think more about this here.

But one way would be to create a client instance. And then this instance needs to be passed when calls are made.